assets/src/dashboard/components/cardGridItem/cardPreview.js

```diff
@@ -80,47 +74,57 @@ const CtaContainer = styled.div`
   margin: auto auto 25%;
 `;
 
-const PlayArrowIcon = styled(PlayArrowSvg).attrs({ width: 11, height: 14 })`
-  margin-right: 9px;
-`;
+const BottomActionButton = styled(Button)``;
 
-const EditButton = styled(Button).attrs({ onClick: () => {} })``;
+const getActionAttributes = (targetAction) =>
+  typeof targetAction === 'string'
+    ? { forwardedAs: 'a', href: targetAction, onClick: () => {} }
+    : { onClick: targetAction };
 
-// TODO modify to handle other types of grid items, not just own stories
-const CardPreviewContainer = ({ editUrl, onPreviewClick, children }) => {
-  const displayEditControls = onPreviewClick || editUrl;
+const CardPreviewContainer = ({ centerAction, bottomAction, children }) => {
   const { pageSize } = usePagePreviewSize();
+
   return (
     <>
       <PreviewPane cardSize={pageSize}>{children}</PreviewPane>
-      {displayEditControls && (
-        <EditControls cardSize={pageSize}>
-          {onPreviewClick && (
-            <PreviewContainer>
-              <PreviewButton
-                type={BUTTON_TYPES.SECONDARY}
-                onClick={onPreviewClick}
-              >
-                <PlayArrowIcon />
-                {__('Preview', 'web-stories')}
-              </PreviewButton>
-            </PreviewContainer>
-          )}
-          <CtaContainer>
-            <EditButton forwardedAs="a" href={editUrl}>
-              {__('Open in editor', 'web-stories')}
-            </EditButton>
-          </CtaContainer>
-        </EditControls>
-      )}
+      <EditControls cardSize={pageSize}>
+        {centerAction && (
+          <CenterActionContainer>
+            <CenterActionButton
+              type={BUTTON_TYPES.SECONDARY}
+              {...getActionAttributes(centerAction.targetAction)}
+            >
+              {centerAction.label}
+            </CenterActionButton>
+          </CenterActionContainer>
+        )}
+        <CtaContainer>
+          <BottomActionButton
+            {...getActionAttributes(bottomAction.targetAction)}
+          >
+            {bottomAction.label}
+          </BottomActionButton>
+        </CtaContainer>
+      </EditControls>
     </>
   );
 };
 
+export const ActionLabel = PropTypes.oneOfType([
+  PropTypes.string,
+  PropTypes.node,
+]);
```

Comment:
should this go in dasbhoard/types.js instead?

Reply:
Hmmm... nice callout, but I kept the definition local because it's a custom prop that's really specific to a `cardGridItem`, I'm not sure how generically it can be used like the propTypes in the global `types` files.

However, to keep things clean and not clutter things up, I created a local `types.js` file inside the `cardGridItem` folder and moved the definition there, so we can have local types easily identified, and if ever a type needs to be moved from local to global, it'd be a quick copy and paste up the tree.